### PR TITLE
Add basic Sandbox data continuity tests

### PR DESCRIPTION
**Pull request by cocreature**
This adds some basic tests that check that data migrations work
properly. For now, I use DAML Script to create and query contracts at
each step. This isn’t perfect since queries can only use the active
contract service but not things like the transaction stream but it’s
clearly better than nothing.

The runner for executing the tests is a simple Haskell executable. It
didn’t really seem useful to throw tasty at this.

I’ve added two sets of tests, one that runs only through stable
versions and one that includes snapshots since migrating through
snapshots is not necessarily equivalent.

Sadly these tests use sandbox-classic since I discovered while writing
these tests that sandbox-next does not actually support migrating data
between SDK versions.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compatibility/sandbox-migration/SandboxMigrationRunner.hs [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+instance A.FromJSON Result
+
+withSandbox :: FilePath -> T.Text -> (Int -> IO a) -> IO a
+withSandbox assistant jdbcUrl f =
+    withTempFile $ \portFile ->
+    withCreateProcess (proc assistant (args portFile)) { create_group = True } $ \_ _ _ ph -> do
+        p <- readPortFile maxRetries portFile
+        r <- f p
+        -- This is a shell process so we need to kill the whole process group.
+        interruptProcessGroupOf ph
```
Comment (cocreature) [inline compatibility/sandbox-migration/SandboxMigrationRunner.hs:168]: I haven’t looked too closely so far but I expect we probably have to fix this in the ledger tests as well. I’ll leave that for a separate PR.

Comment (cocreature) [reply compatibility/sandbox-migration/SandboxMigrationRunner.hs]: I ended up fixing the `Sandbox` module in this PR and switched over to this.

File: compatibility/sandbox-migration/SandboxMigrationRunner.hs [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+-- This test runs through the following steps:
+-- 1. Start postgres
+-- 2. Start the oldest version of sandbox.
+-- 3. Upload a DAR using the same SDK version.
+-- 4. Stop sandbox.
+-- 5. In a loop over all versions:
+--    1. Start sandbox of the given version.
+--    1. Run a script for querying and creating new contracts.
+--    2. Stop sandbox.
```
Comment (aherrmann-da) [inline compatibility/sandbox-migration/SandboxMigrationRunner.hs:16]:
```suggestion
--    2. Run a script for querying and creating new contracts.
--    3. Stop sandbox.
```

Comment (cocreature) [reply compatibility/sandbox-migration/SandboxMigrationRunner.hs:16]: Counting to 3 seems to be too hard for me :facepalm:
